chia/rpc/full_node_rpc_api.py

```diff
@@ -17,6 +17,8 @@
 from chia.util.ints import uint32, uint64, uint128
 from chia.util.log_exceptions import log_exceptions
 from chia.util.ws_message import WsRpcMessage, create_payload_dict
+from chia.wallet.nft_wallet.nft_info import NFTInfo
+from chia.wallet.nft_wallet.nft_puzzles import get_nft_info_from_puzzle
```

Comment:
I'm not sure `full_node_rpc_api.py` is the right place for this RPC, given that it doesn't currently import anything from `chia.wallet`. There may be a reason for keeping the full node and wallet RPC implementations separate. @mariano54 do you have any objections to adding a dependency on wallet code here?

I can see how this RPC sort of fits with the other similar `get_coin_records_*` RPCs, so I understand the logic behind including it here. Was there some other reason for not including this RPC in `wallet_rpc_api.py`?

Reply:
There should be no wallet specific code in the full node. This doesn't belong here

Reply:
There are many many reasons for this but TLDR is modularity